rands.c

```diff
@@ -60,137 +53,20 @@ void RandSeed(signed long seed) {
 
 /*****************************************************/
 /**
-  \fn double RandUni_fast(void)
-  \brief Generate a uniform random variate.
+  Random number generator using the specified rng.
 
-  "Fast" because it
-  utilizes the system rand() but shuffles the results
-  to make a less correlated sequence.  This code is
-  based on FUNCTION RAN0 in Press, et al., 1986,
-  Numerical Recipes, p196, Press Syndicate, NY.
-
-  Of course, just how fast is "fast" depends on the
-  implementation of the compiler.  Some older generators
-  may be quite simple, and so would be faster than
-  a more complicated algorithm.  Newer rand()'s
-  are often fast and good.
-
-  cwb 18-Dec-02
-
-	\return double. A value between 0 and 1.
+  \parameters: pcg32_random_t* that specifies which stream to use.
+ 
+  \returns: a double between 0 and 1.
 */
-double RandUni_fast(void) {
-	static double y;
-
-	#ifdef RSOILWAT

```

Comment:
You cannot deleted this `#ifdef RSOILWAT` case -- c code compiled for the R package `rSOILWAT2` must use the R API RNGs (see https://cran.r-project.org/doc/manuals/R-exts.html#Writing-portable-packages and https://cran.r-project.org/doc/manuals/R-exts.html#Random-numbers).